programs/keeper/TinyContext.cpp

```diff
@@ -31,7 +31,7 @@ void TinyContext::initializeKeeperDispatcher([[maybe_unused]] bool start_async)
     std::lock_guard lock(keeper_dispatcher_mutex);
 
     if (keeper_dispatcher)
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Trying to initialize Keeper multiple times");
+        throw Exception(ErrorCodes::KEEPER_EXCEPTION, "Trying to initialize Keeper multiple times");
```

Comment:
The reason i changed the error type is, when LOGICAL_ERROR happened in function getKeeperDispatcher, the debug build would fail directly so that many CI tests would not pass. And in many tests, the keeper server is not the test point so it is not initialized and exception occurs and break tests.
Do you have any suggestion about this?

Reply:
@antonio2368 any thoughts?

Reply:
Again, this shouldn't happen (Keeper initialization multiple times, so the code should be fixed and not the exception) so I would leave the `LOGICAL_ERROR` here.

Reply:
I see. Using LOGICAL_ERROR back. But removed the exception in getKeeperDispatcher function to make caller handle the null return value case.

Reply:
Now you need to add the checks everywhere where you call that function and for most of them you need to throw `LOGICAL_ERROR` if it's not initialized.
I would add `tryGetKeeperDispatcher` for your usecase.
WDYT?

Reply:
I checked all the caller functions which use the `getKeeperDispatcher`, and before the function call, `initializeKeeperDispatcher` has already been called. 
But okay, in case there is other caller in future, i will add tryGetKeeperDispatcher instead in my case and keep the original logic.